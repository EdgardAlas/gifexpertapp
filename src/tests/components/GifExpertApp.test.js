import GifExpertApp from "../../GifExpertApp"
const { shallow } = require("enzyme/build");

describe('Pruebas en GifExpertApp', () => {
    test('Match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ["one punch", "dragon ball"]
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GifGrid").length).toBe(categories.length)
    })

});
