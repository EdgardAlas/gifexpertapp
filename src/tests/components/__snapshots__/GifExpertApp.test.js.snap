// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ul>
</Fragment>
`;
